pdns/pdnsutil.cc

```diff
@@ -584,16 +584,16 @@ int checkZone(DNSSECKeeper &dk, UeberBackend &B, const DNSName& zone, const vect
       if (rr.qtype.getCode() == QType::SRV) {
         vector<string> parts;
         stringtok(parts, rr.getZoneRepresentation());
-        toCheck = DNSName(parts[3]);
+        if (parts.size() == 4) toCheck = DNSName(parts[3]);
       } else if (rr.qtype.getCode() == QType::MX) {
         vector<string> parts;
         stringtok(parts, rr.getZoneRepresentation());
-        toCheck = DNSName(parts[1]);
+        if (parts.size() == 2) toCheck = DNSName(parts[1]);
       } else {
         toCheck = DNSName(rr.content);
       }
 
-      if (!toCheck.isHostname()) {
+      if (!toCheck.empty() && !toCheck.isHostname()) {
```

Comment:
Perhaps we should display a warning if the hostname is empty? Something like:

` if (toCheck.empty() || !toCheck.isHostname()) {`

Reply:
I pondered that, but I couldn't imagine a code path that ends with an empty `toCheck` without having given some other warning before. But it doesn't hurt to check, so I'll fix it.

Reply:
pushed!